### PR TITLE
Abductor Augments + Organ cleanup

### PR DESCRIPTION
**Pull request by RemieRichards**
- Framework for Augments to have different icon files and icon states (augment_icon_state is the only one used atm, augment_icon was just easy to implement just incase the .dmi fills up too much)
- Variable to prevent `/robot_parts` from being used in cyborgs: `usecyborg`
- Colons `:` removed from robot_parts.dm
- `/robot_parts` now have a body_part var like organs, this is now used instead of a switch and 6 typechecks to prevent incorrect augs being attached to incorrect limbs.
- Robotic organs no longer bleed
- Augmentation no longer deletes the limb and replaces it with a robot subtype, it simply sets `status == ORGAN_ROBOTIC` and repairs any damage on the limb (it's a "new" limb fluff wise, so damage is removed) `ORGAN_ROBOTIC` was used for the majority of robotic limb checks already and mimics my dismemberment changes.
- Abductor augments! Sprites by @Ausops (Not mapped in, I can't be arsed to map or mapmerge)

alienaug

### CODE REVIEW

**Comment by AlexanderUlanH** [comment]
>Augmented limbs are no longer object-based

It makes me sad, but if the only difference between the limbs was the flag we might as well.

**Comment by RemieRichards** [comment]
@AlexanderUlanH they're still object based technically! (Organs are objects, have been for ages, it's an old refactor of mine) plus they'd be moved to the flag system fully once dismemberment went in anyway.

**Comment by AnturK** [comment]
We need peg leg and hook hand icons now. :+1:

**Comment by RemieRichards** [comment]
@AnturK move augment_icon and augment_icon_state down to /item and change the surgery, done! honk. (Maybe once I've made the damage the limb prevents a variable, not right now)

**Comment by Miauw62** [comment]
If you plan on having specific organ effects in the future (robolungs and stuff) you probably want to keep roborgans as objs

**Comment by RemieRichards** [comment]
@Miauw62 the specific effects will only ever be icons (this PR), punchdamage (that's one var) and damage reduction (that's one var)
Everything else is covered by the status var already, or can use it.

**Comment by Miauw62** [comment]
yeah but \*futureproofing\*

**Comment by RemieRichards** [comment]
@Miauw62 I've just explained all the "futureproofing"

**Comment by WJohn** [comment]
These are some nice-ass sprites.

<3 ausops.

**Comment by GunHog** [comment]
Post screenshots, please?

**Comment by RemieRichards** [comment]
@GunHog op.

**Comment by Man-Shroom** [comment]
Oh god oh jeez is this code getting cleaned up so dismemberment will be easier? Hallelujah thank you jesus

File: code/game/gamemodes/abduction/abduction_gear.dm [hunk 1]
```diff
@@ -557,4 +557,56 @@ Congratulations! You are now trained for xenobiology research!"}
...
-	icon_door = "abductor"
\ No newline at end of file
+	icon_door = "abductor"
+
+
+
+//Abductor augs
+
+/obj/item/robot_parts/abductor
+	name = "aliem robotic part"
```
Comment (HoaxIT) [inline code/game/gamemodes/abduction/abduction_gear.dm]: AYYYYYYYLIEEMMMMSSS

Comment (Dorsisdwarf) [inline code/game/gamemodes/abduction/abduction_gear.dm]: Pretty sure it's a parent part and doesnt exist ingame

Comment (RemieRichards) [inline code/game/gamemodes/abduction/abduction_gear.dm]: yeah, not worth fixing the typo to be honest, if I have to commit something else I'll fix it, but it's unseen text, it's fine.

**Comment by Razharas** [comment]
@RemieRichards dont let the fate of dismemberment fall upon this pr

**Comment by SconesC** [comment]
the fate of dismemberment has fallen upon this pr

**Comment by Razharas** [comment]
Indeed
Reopen if you will return to it

**Pull request closed without merging**